### ISSUE TITLE
Compose Version Doesnt Work//Multi-Path Handling Bug

### ISSUE THREAD

**Issue by Renderer6060** [Salvoxia/immich-folder-album-creator#52]
Hi 

Ive spent a good few hours trying to work out why this doesnt work but im at a loss - All ive managed to find is the .sh version ran with arguments in the compose version doesnt parse the same as the python version for some reason.

Here is the issue 

When I run the below - it detects albums to be made:
docker run -e API_URL="http://MY>IP:2283/api" -e API_KEY="MY_API_KEY" -e ALBUM_LEVELS="4" -e ALBUM_SEPARATOR=" - " -e ROOT_PATH="/usr/src/app/external/Photos" salvoxia/immich-folder-album-creator:latest /script/immich_auto_album.sh

it spits out this:
time=2024-09-21T17:56:28.753+00:00 level=INFO msg=Detected Immich server version 1.115.0
time=2024-09-21T17:56:28.753+00:00 level=INFO msg=Requesting all assets
time=2024-09-21T17:56:28.781+00:00 level=INFO msg=34 photos found
time=2024-09-21T17:56:28.781+00:00 level=INFO msg=Sorting assets to corresponding albums using folder name
time=2024-09-21T17:56:28.782+00:00 level=INFO msg=11 albums identified
time=2024-09-21T17:56:28.782+00:00 level=INFO msg=Album list: ['Other - Logos' OTHER ABLUMBS ETC]

So i thought great, I wont run it unattended, ill put that in a compose file as per the docs and as below 

  immich-folder-album-creator:
    container_name: immich_folder_album_creator
    image: salvoxia/immich-folder-album-creator:latest
    restart: unless-stopped
    environment:
      API_URL: http://MY_IP:2283/api
      API_KEY: MY_API_KEY
      ALBUM_LEVELS: 4
      ALBUM_SEPARATOR: " - "
      SYNC_MODE: 2
      ROOT_PATH: /usr/src/app/external/Photos
      CRON_EXPRESSION: "*0 * * * *"
      TZ: Europe/London

however, on the hour, every hour - the container output is this:

time=2024-09-21T19:15:00.655+01:00 level=INFO msg=Detected Immich server version 1.115.0

time=2024-09-21T19:15:00.656+01:00 level=INFO msg=Requesting all assets

time=2024-09-21T19:15:00.670+01:00 level=INFO msg=0 photos found

time=2024-09-21T19:15:00.670+01:00 level=INFO msg=Sorting assets to corresponding albums using folder name

time=2024-09-21T19:15:00.670+01:00 level=INFO msg=0 albums identified

time=2024-09-21T19:15:00.671+01:00 level=INFO msg=Album list: []

time=2024-09-21T19:15:00.671+01:00 level=INFO msg=Listing existing albums on immich

Makes no sense - its not seeing the albums that the exact same command ran manually can see.

This tool has already been invaluable to me in my immich migration, and for now i can just run it manually on an interval but id love to get this working for some better automation so some help would be great

Thanks

**Comment by Salvoxia** [comment]
Hi,

I'm sorry you're having difficulties setting it up and you wasted so much time, I hope we can get you going!

The output 
```
time=2024-09-21T17:56:28.781+00:00 level=INFO msg=34 photos found
```
is the raw number of assets the API has returned, BEFORE evaluating any root paths.
The fact that it says
```
time=2024-09-21T19:15:00.670+01:00 level=INFO msg=0 photos found
```
when running from compose indicates that it did indeed not receive any assets from the API.

Since you seem to have read the documentation and you used `SYNC_MODE: 2` in your docker compose file, you also know that you must use an Admin user's API key for `SYNC_MODE: 2` to work. So here's my question:  
When running the command manually and when setting up the docker compose file, did you use the same API key for the same Admin user? If you used a different user's API key during the manual run you would have seen that user's assets. If the Admin user does not have any assets, that would explain it.

Can you shed some light on this?

Best Regards,
Salvoxia

**Comment by Salvoxia** [comment]
In your latest compose stack you seem to use two `ROOT_PATH` entries:
```yaml
ROOT_PATH: "/usr/src/app/external/Photos,/usr/src/app/external/Video"
```

Just to make sure that this is not what's causing the issues, could you try the compose stack with the standard API key and just the single `ROOT_PATH` used in your manual command?

**Comment by Salvoxia** [comment]
> That appears to have been it - that was a copy/paste typo from a test branch I was running. Using the user non-admin key, removing SYNC and running a single file path seems to have solved it.

That's not right then, maybe there actually is a problem with multiple `ROOT_PATH` handling.

> Out of interest - could the issue with SYNC=2 and the admin key be that the admin key belonged to my Admin user, but the external library was set up for use by a non-admin?

Yes, sorry, I should have explained it better. Since API Keys are tied to a user and the external library is also tied to a user, the script can only see what the respective user can see. If the user does not have an external library, the script cannot see it.

**Comment by Salvoxia** [comment]
Thanks for the kind words!

I just pushed a fix for handling multiple root paths to the [dev](https://github.com/Salvoxia/immich-folder-album-creator/tree/dev) branch which builds the `edge` Docker tag. 
I'd appreciate it if you could test whether multiple root paths are now working as expected for you with the `edge` tag.

**Comment by Salvoxia** [comment]
Thanks for testing!

The fix is now released with [v0.11.1](https://github.com/Salvoxia/immich-folder-album-creator/releases/tag/0.11.1).
